Support empty tables

Sohalt opened the pull request:
Shows "no data" for empty tables, instead of an error.

![2024-07-30-183954_985x193_scrot](https://github.com/user-attachments/assets/2bae4003-eb95-4ba6-bef6-7a063b860f19)

Alternatively just show nothing?

Closes #661

mk commented:
What does it look like for a table with only a `:head` and empty `:rows`? I'd try something more descriptive like `this table has no rows` in the message.

Sohalt commented:
> What does it look like for a table with only a `:head` and empty `:rows`?

It would look the same as for a completely empty table. I changed it to still show the headers. I also changed the message.

It now looks like this:

![2024-07-31-101225_907x493_scrot](https://github.com/user-attachments/assets/e8919e6c-6c5e-420a-a255-372c1e4d4280)

Not entirely sure about the `:nextjournal/width` default. For tables it defaults to `:wide`, but with the empty table that looks a bit weird imo. I also think it would be nice to have the "this table has no rows" message centered in the table, but I couldn't get that to work :/

mk commented:
@Sohalt cool, thanks. Can you add a changelog entry and then we merge it?
